test/react/functional-components/refs2.js

```diff
@@ -0,0 +1,32 @@
+var React = require('react');
+// the JSX transform converts to React, so we need to add it back in
+this['React'] = React;
+
+function Child(props) {
+  return <div ref={props.forwardedRef} />;
+}
+
+const WrappedComponent = React.forwardRef((props, ref) => {
+  return <Child {...props} forwardedRef={ref} />;
+});
+
+function App(props) {
+  return (
+    <WrappedComponent ref={props.x} />
+  );
+}
+
+App.getTrials = function(renderer, Root) {
+  let results = [];
+  var refValue = null;
+  renderer.update(<Root x={val => refValue = val} />);
+  results.push(['simple render with refs', renderer.toJSON()]);
+  results.push(['ref node is of div', refValue.type]);
```

Comment:
Can we also verify that:

* You can forward refs more through several levels
* It works with class instances too (e.g. test `ref.current.getValue()` with `getValue() { return this.props.value; }`)

Reply:
- You can forward refs more through several levels

Can you give me an example of a test like this? Ideally, if you could just add a said test to this PR that would be awesome.

Reply:
`forwardRef` that renders another `forwardRef`

Reply:
Maybe this doesn't make sense. I'm not sure.